### PR TITLE
ref(integrations): Simplify More Notifications Tests

### PR DESCRIPTION
**Pull request by mgaeta**
The Dual Write Migration is still too large after splitting into multiple PRs. This PR isolates refactors of tests files that are pretty verbose but change no logic. The two main themes are:
- Use `get_valid_response`
- DRY repeated logic into helper functions

### CODE REVIEW

File: tests/sentry/api/endpoints/test_project_details.py [hunk 1]
```diff
@@ -147,50 +133,40 @@ def test_non_org_rename_403(self):
...
+        self.proj_slug = self.project.slug
         self.login_as(user=self.user)
 
     def test_blank_subject_prefix(self):
         project = Project.objects.get(id=self.project.id)
         options = {"mail:subject_prefix": "[Sentry]"}
-        resp = self.client.put(self.path, data={"options": options})
-        assert resp.status_code == 200, resp.content
+
+        self.get_valid_response(self.org_slug, self.proj_slug, options=options)
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_project_details.py:155]: doesn't this need the `method=put`?

File: tests/sentry/api/endpoints/test_project_details.py [hunk 2]
```diff
@@ -147,50 +133,40 @@ def test_non_org_rename_403(self):
...
         assert project.get_option("mail:subject_prefix") == "[Sentry]"
 
         options["mail:subject_prefix"] = ""
-        resp = self.client.put(self.path, data={"options": options})
-        assert resp.status_code == 200, resp.content
+        self.get_valid_response(self.org_slug, self.proj_slug, options=options)
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_project_details.py:159]: same here

File: tests/sentry/api/endpoints/test_project_details.py [hunk 3]
```diff
@@ -378,193 +365,180 @@ def test_options(self):
...
+        self.get_valid_response(self.org_slug, self.proj_slug, isBookmarked="false")
         assert not ProjectBookmark.objects.filter(
             project_id=self.project.id, user=self.user
         ).exists()
 
     def test_subscription(self):
-        resp = self.client.put(self.path, data={"isSubscribed": "true"})
-        assert resp.status_code == 200, resp.content
+        self.get_valid_response(self.org_slug, self.proj_slug, isSubscribed="true")
+        # TODO MARCOS
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_project_details.py:375]: was this leftover? what's the todo?

File: tests/sentry/api/endpoints/test_project_details.py [hunk 4]
```diff
@@ -378,193 +365,180 @@ def test_options(self):
...
+        resp = self.get_valid_response(self.org_slug, self.proj_slug, relayPiiConfig=value)
         assert self.project.get_option("sentry:relay_pii_config") == value
         assert resp.data["relayPiiConfig"] == value
 
     def test_sensitive_fields(self):
-        resp = self.client.put(
-            self.path, data={"sensitiveFields": ["foobar.com", "https://example.com"]}
-        )
-        assert resp.status_code == 200, resp.content
+        value = ["foobar.com", "https://example.com"]
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_project_details.py:493]: not a big deal but why did you assign this as a var when in other places (like line 473) you don't? I think consistency is good

File: tests/sentry/api/endpoints/test_project_details.py [hunk 5]
```diff
@@ -623,10 +599,9 @@ def test_setting_dynamic_sampling_rules_roundtrip(self):
         Tests that we get the same dynamic sampling rules that previously set
         """
         with Feature({"organizations:filters-and-sampling": True}):
-            resp = self.client.put(self.path, data={"dynamicSampling": _dyn_sampling_data()})
-            assert resp.status_code == 200, resp.content
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_project_details.py:627]: this one seems like it's missing a replacement

File: tests/sentry/api/endpoints/test_project_details.py (hunk 1 above)
Comment (ceorourke) [reply tests/sentry/api/endpoints/test_project_details.py:155]: nvm I see it above

File: tests/sentry/api/endpoints/test_user_details.py [hunk 6]
```diff
@@ -1,79 +1,55 @@
-from django.core.urlresolvers import reverse
-
 from sentry.models import Organization, OrganizationStatus, User, UserOption
 from sentry.testutils import APITestCase
 
 
 class UserDetailsTest(APITestCase):
-    # TODO(dcramer): theres currently no way to look up other users
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_user_details.py:8]: does this no longer apply? if it does or you're not sure, better to keep it

File: tests/sentry/api/endpoints/test_user_details.py [hunk 7]
```diff
@@ -133,8 +111,7 @@ def test_change_username_when_different(self):
         user = self.create_user(email="c@example.com", username="diff@example.com")
         self.login_as(user=user, superuser=False)
 
-        resp = self.client.put(self.url, data={"username": "new@example.com"})
-        assert resp.status_code == 200, resp.content
+        self.get_valid_response("me", username="new@example.com")
```
Comment (ceorourke) [inline tests/sentry/api/endpoints/test_user_details.py:114]: what is "me" being passed in for?

**Review by ceorourke — COMMENTED** [review]
overall looks good just left some comments

File: tests/sentry/api/endpoints/test_user_details.py (hunk 7 above)
Comment (mgaeta) [reply tests/sentry/api/endpoints/test_user_details.py:114]: https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/user.py#L56
When the base user endpoint gets arguments from the URL, it knows to replace `"me"` with `request.user.id`. This isn't explained anywhere so I'm going to add a line of documentation.

File: tests/sentry/api/endpoints/test_user_details.py (hunk 6 above)
Comment (mgaeta) [reply tests/sentry/api/endpoints/test_user_details.py:8]: It's hard to maintain code that's hidden in comments. The original version still lives in git history so I usually try to delete them when I see them. You're right though, I shouldn't remove the TODO.

File: tests/sentry/api/endpoints/test_project_details.py (hunk 3 above)
Comment (mgaeta) [reply tests/sentry/api/endpoints/test_project_details.py:375]: Yeah leftover from splitting the PR up.

File: tests/sentry/api/endpoints/test_project_details.py (hunk 5 above)
Comment (mgaeta) [reply tests/sentry/api/endpoints/test_project_details.py:627]: Thanks! I looks like there was a bug that was making this test still pass.

**Comment by mgaeta** [comment]
Ok this is passing all checks now.

**Review by ceorourke — APPROVED** [review]
looks good!

**Pull request merged**